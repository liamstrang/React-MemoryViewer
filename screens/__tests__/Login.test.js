import React from 'react';
import renderer from 'react-test-renderer';
import Login from '../Login'

test('Login Snapshop', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
});

describe('Check Login Page', () => {
    it('Render Login Page', () => {
        const tree = renderer.create(<Login />).toJSON();
        expect(tree.children[2].type).toBe('View')
    })
    it('Render Login Image', () => {
        const tree = renderer.create(<Login />).toJSON();
        expect(tree.children[0].props.source).toBe(1)
    })
    it('Login Styling', () => {
        const json = renderer.create(<Login/>).toJSON()
        expect(json.props.style[0].width).toBe('100%')
        expect(json.props.style[0].flex).toBe(1)
        expect(json.props.style[0].maxWidth).toBe(340)
        expect(json.props.style[0].padding).toBe(20)
        expect(json.props.style[0].alignItems).toBe('center')
        expect(json.props.style[0].justifyContent).toBe('center')
    })
})