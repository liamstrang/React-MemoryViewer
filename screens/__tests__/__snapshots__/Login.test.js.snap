// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Snapshop 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flex": 1,
        "justifyContent": "center",
        "maxWidth": 340,
        "padding": 20,
        "width": "100%",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 128,
        "marginBottom": 12,
        "width": 128,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#004ae6",
        "fontSize": 26,
        "fontWeight": "bold",
        "paddingVertical": 14,
      }
    }
  >
    Login
  </Text>
  <View
    style={
      Object {
        "marginVertical": 12,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "marginVertical": 12,
        "width": "100%",
      }
    }
  />
</View>
`;
